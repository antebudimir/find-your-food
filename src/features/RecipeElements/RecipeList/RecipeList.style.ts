import styled from 'styled-components';

export const RecipeListContainer = styled.ul``;

export const RecipeListText = styled.span``;
