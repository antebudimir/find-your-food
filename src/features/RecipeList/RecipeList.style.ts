import styled from 'styled-components';

export const RecipeListContainer = styled.ul`
	margin-top: ${(props) => props.theme.spacing.primary};
`;

export const RecipeListText = styled.span`
	font-family: 'Raleway Bold';
`;
