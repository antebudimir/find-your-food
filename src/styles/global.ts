import { createGlobalStyle } from 'styled-components';
import RalewayRegular from 'fonts/Raleway-Regular.woff';
import RalewayBold from 'fonts/Raleway-Bold.woff';
import RalewayExtraBold from 'fonts/Raleway-ExtraBold.woff';

export const GlobalStyles = createGlobalStyle`
@font-face {
	font-family: 'Raleway Regular';
	src: url(${RalewayRegular}) format('woff');
	font-weight: normal;
	font-style: normal;
	font-display: swap;
}

@font-face {
	font-family: 'Raleway Bold';
	src: url(${RalewayBold}) format('woff');
	font-weight: bold;
	font-style: normal;
	font-display: swap;
}


@font-face {
    font-family: 'Raleway Extra Bold';
    src: url(${RalewayExtraBold}) format('woff');
    font-weight: bold;
    font-style: normal;
    font-display: swap;
}

* {
	outline: none;
	box-sizing: border-box;
	margin: 0;
	padding: 0;
	scroll-behavior: smooth;
}

html {
    font-family: 'Raleway Regular';
	word-wrap: break-word;
}

button,
input {
	border: none;
	display: block;
}

input,
select,
textarea,
button {
	color: inherit;
	font-size: inherit;
}

input,
select,
textarea {
	font-family: inherit;
}

h1 {
    font-family: 'Raleway Extra Bold';
}

h2,
button,
a {
	font-family: 'Raleway Bold';
}

ul,
li {
	list-style: none;
}

a {
	text-decoration: none;
}

img {
	display: block;
	max-width: 100%;
}
`;
