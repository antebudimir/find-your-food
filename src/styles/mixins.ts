import { css } from 'styled-components';

interface FlexLayoutProps {
	direction?: string;
	wrap?: string;
}

interface FlexAlignmentProps {
	justify?: string;
	align?: string;
}

export const FlexLayout = ({ direction, wrap }: FlexLayoutProps) => css`
	display: flex;
	flex-flow: ${direction} ${wrap};
`;

export const FlexAlignment = ({ justify, align }: FlexAlignmentProps) => css`
	justify-content: ${justify};
	align-items: ${align};
`;

export const hiddenVisually = css`
	position: absolute;
	clip: rect(0 0 0 0);
	clip-path: inset(50%);
	overflow: hidden;
	width: 1px;
	height: 1px;
	white-space: nowrap;
`;

export const sticky = css`
	display: flex;
	flex-flow: column nowrap;
	min-height: 89.6vh;
`;
