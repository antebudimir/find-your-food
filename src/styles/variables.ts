export const variables = {
	borderRadius: {
		primary: '0.125rem',
	},
	outline: {
		primary: '1px',
	},
	shadow: {
		primary: '#3b3f20 0px 2px 5px 0px',
		secondary: '#d45464 0px 2px 5px 0px',
		tertiary: '5px 5px 20px rgba(0, 0, 0, 0.7)',
		quaternary: '5px 5px 5px rgba(0, 0, 0, 0.7)',
	},
	sizing: {
		inputDesktopWidth: '28rem',
		summaryLinkWidth: '18.9rem', // coresponds to the width of the image
		summaryContainerWidth: '28rem',
		mainContainerWidth: '60rem',
	},
	spacing: {
		primary: '1rem',
		secondary: '0.5rem 1rem',
		tertiary: '2rem',
		quaternary: '1.5rem',
	},
	color: {
		primary: '#d45464',
		primaryLighter: '#ff6478',
		secondary: ' #ffffff',
		tertiary: '#3b3f20',
	},
	fontSize: {
		primary: '1rem',
		secondary: '1.3rem',
		tertiary: '1.5rem',
		quaternary: '2rem',
		quinary: '120%',
	},
	timing: {
		primary: '200ms',
	},
	lineHeight: {
		primary: '1.5',
		secondary: '2',
	},
};
