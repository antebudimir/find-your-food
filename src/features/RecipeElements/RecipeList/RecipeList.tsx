import { RecipeListContainer, RecipeListText } from './RecipeList.style';
import { ReactNode } from 'react';

interface RecipeListProps {
	label?: string;
	detailsList?: string | ReactNode[];
}

const RecipeList = ({ label = '', detailsList = '' }: RecipeListProps) => {
	return (
		<RecipeListContainer>
			<>
				<RecipeListText>{label}:</RecipeListText> {detailsList}
			</>
		</RecipeListContainer>
	);
};

export default RecipeList;
