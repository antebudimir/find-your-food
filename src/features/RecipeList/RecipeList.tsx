import { RecipeListContainer, RecipeListText } from './RecipeList.style';
import { ReactNode } from 'react';

export interface RecipeListProps {
	label?: string;
	detailsList?: string | ReactNode[];
}

const RecipeList = ({ label, detailsList }: RecipeListProps) => {
	return (
		<RecipeListContainer data-testid="recipe-list">
			<RecipeListText>{label ? label : 'Title not available'}:</RecipeListText>{' '}
			{detailsList ? detailsList : ''}
		</RecipeListContainer>
	);
};

export default RecipeList;
