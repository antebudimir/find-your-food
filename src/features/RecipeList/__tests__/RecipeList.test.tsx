import { ThemeProvider } from 'styled-components';
import { variables } from 'styles/variables';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import RecipeList, { RecipeListProps } from '../RecipeList';

const props: RecipeListProps = {
	label: 'Test label',
	detailsList: 'Test detail',
};

const renderRecipeList = (props: RecipeListProps) => {
	const wrapper = document.createElement('section');
	wrapper.classList.add('section-wrapper');

	const { asFragment } = render(
		<ThemeProvider theme={variables}>
			<RecipeList {...props} />
		</ThemeProvider>,
		{ container: document.body.appendChild(wrapper) },
	);

	return { fragment: asFragment() };
};

describe('<RecipeList /> component', () => {
	test('if props are being passed', () => {
		renderRecipeList(props);

		const recipeList = screen.getByTestId('recipe-list');
		expect(recipeList).toHaveTextContent('Test label: Test detail');
	});

	test('if the <RecipeList /> matches the snapshot', () => {
		const { fragment } = renderRecipeList(props);
		expect(fragment).toMatchSnapshot();
	});
});
