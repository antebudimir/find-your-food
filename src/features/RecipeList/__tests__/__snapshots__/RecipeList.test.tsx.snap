// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> component if the <RecipeList /> matches the snapshot 1`] = `
<DocumentFragment>
  <ul
    class="sc-bczRLJ hHaKtq"
    data-testid="recipe-list"
  >
    <span
      class="sc-gsnTZi iqEmM"
    >
      Test label:
    </span>
     Test detail
  </ul>
</DocumentFragment>
`;
